Source/DafnyLanguageServer/Workspace/DafnyDocument.cs

```diff
@@ -26,8 +27,8 @@ public record DafnyDocument(
     IReadOnlyList<Diagnostic> ParseAndResolutionDiagnostics,
     // VerificationDiagnostics can be deduced from CounterExamples,
     // but they are stored separately because they are migrated and counterexamples currently are not.
-    IReadOnlyDictionary<ImplementationId, ImplementationView>? ImplementationViews,
-    IReadOnlyList<Counterexample> CounterExamples,
+    IReadOnlyDictionary<ImplementationId, ImplementationView>? ImplementationViewsView,
```

Comment:
In a future PR I plan to refactor `DafnyDocument`, so there is one mutable version which I'll rename to `Compilation`, and one immutable version which is a `CompilationSnapshot` (or something) that is used for publishing information. It's useful to have an immutable object so that when this object is on its way to be published, it is not mutated by ongoing processes, which is especially useful for reliable testing.